geonode/upload/views.py

```diff
@@ -707,15 +707,25 @@ def view(req, step=None):
                 return error_response(req, errors=e.args)
 
         resp = _steps[step](req, upload_session)
+
+        try:
+            content = resp.content
+            if isinstance(content, bytes):
+                content = content.decode('UTF-8')
+            resp_js = json.loads(content)
+
+            if not settings.ASYNC_SIGNALS:
+                if 'upload/final' in resp_js.get('redirect_to', ''):
```

Comment:
This looks like a hack. Don't we have a better solution rather then relying on the "redrect_to" content of the response? I know this is a quick solution to move the final step out of the client, but it's quite ugly to me...

Reply:
Well actually this is the check done before by the upload workflow and, as far as I know, it is also the most reliable one.
The response from the final view, in the case everything was good and no further action is needed, will be always a JSON containing the `redirect_to` element.
In other words this is how the current `geoserver.upload` workflow has been implemented. I can see if we can check other variables, but I really don't think that in practice we will benefit from that except maybe code readability.

Reply:
np, not nice code but I see your point.